### PR TITLE
Map gen spawn number

### LINKED ISSUE #1773: Add ability to select the number of players on a generated map
**Wanna have the bug fixed quickly?**
Visit Issue hunt...
[Issue hunt](https://issuehunt.io/r/FAForever/downlords-faf-client)

### PR DESCRIPTION
**Pull request by 1-alex98**
Requires new map gen
Fixes #1773

### CODE REVIEW

**Comment by 1-alex98** [comment]
https://github.com/FAForever/Neroxis-Map-Generator/pull/30

File: src/main/java/com/faforever/client/game/CreateGameController.java [hunk 1]
```diff
@@ -377,7 +393,8 @@ public void onRandomMapButtonClicked() {
 
   public void onGenerateMapButtonClicked() {
     try {
-      mapGeneratorService.generateMap().thenAccept(mapName -> {
+      byte spawnCount = (byte) preferencesService.getPreferences().getGeneratorPrefs().getSpawnCountProperty();
```
Comment (Geosearchef) [inline src/main/java/com/faforever/client/game/CreateGameController.java:396]: In theory this always equals the currently selected value, but it feels a bit wrong. Maybe read this value from the spawnCountSpinner instead?

File: src/main/java/com/faforever/client/map/generator/GenerateMapTask.java [hunk 2]
```diff
@@ -64,7 +64,7 @@ protected Void call() throws Exception {
     ProcessBuilder processBuilder = new ProcessBuilder();
     processBuilder.inheritIO();
     processBuilder.directory(workingDirectory.toFile());
-    processBuilder.command("java", "-jar", generatorExecutableFile.toAbsolutePath().toString(), ".", String.valueOf(seed), version, mapFilename);
+    processBuilder.command("java", "-jar", generatorExecutableFile.toAbsolutePath().toString(), ".", String.valueOf(seed), "0.1.6", mapFilename, "8");
```
Comment (Geosearchef) [inline src/main/java/com/faforever/client/map/generator/GenerateMapTask.java:67]: Is this intentional? You are hardcoding and ignoring the version field. What is the 8 supposed to do?

File: src/main/java/com/faforever/client/map/generator/MapGeneratorService.java [hunk 3]
```diff
@@ -107,9 +108,16 @@ private void deleteGeneratedMaps() {
...
 
-
-  public CompletableFuture<String> generateMap() {
-    return generateMap(queryNewestVersion(), seedGenerator.nextLong());
+  /**
+   * We concatenate the spawnCount on the seed, that's is the only way to keep compatibility. Some day in the future
+   * though I proper way of passing generic options would be appreciated. In the future concatenating(making the seed
+   * number long/bigger) options at the end of the seed is preferred over the bit shift.
+   */
+  public CompletableFuture<String> generateMap(Byte spawnCount) {
```
Comment (Geosearchef) [inline src/main/java/com/faforever/client/map/generator/MapGeneratorService.java:116]: use the primitive type

**Review by Geosearchef — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/map/generator/GenerateMapTask.java (hunk 2 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/map/generator/GenerateMapTask.java:67]: left overs that I did not delete

File: src/main/java/com/faforever/client/map/generator/MapGeneratorService.java (hunk 3 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/map/generator/MapGeneratorService.java:116]: yes sir

File: src/main/java/com/faforever/client/game/CreateGameController.java (hunk 1 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/game/CreateGameController.java:396]: Well I could. Makes no difference...

**Comment by 1-alex98** [comment]
Concerning the funny way of including it in the seed. Atm this is the only way old clients donot break immediately :D I already spok with @Katharsas @shalkya (not sure if thats his github name) @norraxx  about how we fix these compatibility issues in the long term.

**Comment by 1-alex98** [comment]
We want to pin the major version (Semantik Versioning) of the map gen and put warnings if you use a higher major version. Also we want to add a variable options part after the seed so that the client does not really care if new options are in the map name. Atm the client makes expectations about the struture of the map name which are pretty strict.

File: src/main/java/com/faforever/client/map/generator/GenerateMapTask.java (hunk 2 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/map/generator/GenerateMapTask.java:67]: Used to give the gen the number of players via a new arg

**Pull request closed without merging**